pydoctor/driver.py

```diff
@@ -279,6 +303,48 @@ def main(args: Sequence[str] = sys.argv[1:]) -> int:
               "for deep-linking your documentation.",
               file=sys.stderr, flush=True)
 
+
+def _sanitize_options(options: Values) -> None:
+    """
+    Check current command line option and updated them for internal usage.
+
+    For example resolve the default values.
+    """
+    if not options.projectmetadirectory:
+        options.projectmetadirectory = options.projectbasedirectory
+
+    meta = None
+    if options.projectmetadirectory:
+        try:
+            meta = pep517_meta_load(options.projectmetadirectory)
+        except (RuntimeError, FileNotFoundError, NotADirectoryError):
+            """
+            PEP517 raises  when metadata file is not found:
+            * py3.8 - RuntimeError
+            * py3.6 - FileNotFoundError Unix
+            * py3.6 - NotADirectoryError Windows
+
+            It's ok if project has no meta data.
+            """
+
+    if not options.projectversion:
+        if meta:
+            options.projectversion = meta.version
+        else:
+            options.projectversion = '0.1.0.dev0'
+
+    if not options.projectname and meta:
+        options.projectname = meta.metadata['name']
```

Comment:
might be nice to load this from the metadata too:

```suggestion
        options.projectname = meta.metadata['name']
        
        if not options.projecturl and meta:
            options.projecturl = meta.metadata['Home-Page']
```

Reply:
Thanks. I will move all the 'meta' stuff to a separate PR, to also see what other info we can extract.

Reply:
meta code removed from this PR.